### ISSUE TITLE
Add some more pipelining documentatoin

### ISSUE THREAD

**Issue by thomasmoelhave** [thomasmoelhave/tpie#102]
The docs at http://www.madalgo.au.dk/tpie/doc/master/pipelining.html are pretty good but could be improved when it comes to stuff like the passive_sorter and pulling in general. There are some examples but they usually work with more complicated "pull" nodes that also acts as pipe_middles (i.e. things are pushed into them as well). I have added a pretty barebones test app in  6b93d3024b13bf66a68d0dfb00f8c98f6032363e I wonder if that app would be useful as an example in the docs?

**Comment by svendcsvendsen** [comment]
I've moved the example to the apps section and referenced it in the documentation. Furthermore I've added a pull node, passive node and forming pipelines section to the documentation. Recently all the nodes in the pipelining library have been documented and are listed in the documentation. 

I think the documentation is now at a state where it is pretty good. Perhaps, in the future, it would be an idea to write a tutorial style page where a full program is implemented and explained.
